recipe/patches/0001-Add-CMake-files.patch

```diff
@@ -0,0 +1,922 @@
+ src/lib/ngram-context.cc               | 13 +++++-
+ src/lib/ngram-output.cc                |  3 --
+ src/test/CMakeLists.txt                | 42 +++++++++++++++++
+ src/test/disttestsetup.sh              |  0
+ src/test/ngramapply_test.sh            |  0
+ src/test/ngramcount_histograms_test.sh |  0
+ src/test/ngramcount_test.sh            |  0
+ src/test/ngramdistcount_test.sh        |  0
+ src/test/ngramdistmake_test.sh         |  0
+ src/test/ngramdistrand.sh              |  0
+ src/test/ngramdistshrink_test.sh       |  0
+ src/test/ngramdistwittenbell_test.sh   |  0
+ src/test/ngramfracdistmake_test.sh     |  0
+ src/test/ngramfracdistshrink_test.sh   |  0
+ src/test/ngramhisttest-main.cc         |  8 +++-
+ src/test/ngraminfo_test.sh             |  0
+ src/test/ngrammake_test.sh             |  0
+ src/test/ngrammarginalize_test.sh      |  0
+ src/test/ngrammerge_test.sh            |  0
+ src/test/ngramperplexity_test.sh       |  0
+ src/test/ngramprint_test.sh            |  0
+ src/test/ngramrand_test.sh             |  0
+ src/test/ngramrandgen_test.sh          |  0
+ src/test/ngramrandtest-main.cc         | 10 +++-
+ src/test/ngramrandtest.cc              |  5 ++
+ src/test/ngramshrink_test.sh           |  0
+ src/test/ngramsymbols_test.sh          |  0
+ 48 files changed, 332 insertions(+), 22 deletions(-)
+ create mode 100644 .gitattributes
+ create mode 100644 CMakeLists.txt
+ create mode 100644 src/CMakeLists.txt
+ create mode 100644 src/bin/CMakeLists.txt
+ mode change 100755 => 100644 src/bin/ngramdisttrain.sh
+ mode change 100755 => 100644 src/bin/ngramfractrain.sh
+ create mode 100644 src/lib/CMakeLists.txt
+ create mode 100644 src/test/CMakeLists.txt
+ mode change 100755 => 100644 src/test/disttestsetup.sh
+ mode change 100755 => 100644 src/test/ngramapply_test.sh
+ mode change 100755 => 100644 src/test/ngramcount_histograms_test.sh
+ mode change 100755 => 100644 src/test/ngramcount_test.sh
+ mode change 100755 => 100644 src/test/ngramdistcount_test.sh
+ mode change 100755 => 100644 src/test/ngramdistmake_test.sh
+ mode change 100755 => 100644 src/test/ngramdistrand.sh
+ mode change 100755 => 100644 src/test/ngramdistshrink_test.sh
+ mode change 100755 => 100644 src/test/ngramdistwittenbell_test.sh
+ mode change 100755 => 100644 src/test/ngramfracdistmake_test.sh
+ mode change 100755 => 100644 src/test/ngramfracdistshrink_test.sh
+ mode change 100755 => 100644 src/test/ngraminfo_test.sh
+ mode change 100755 => 100644 src/test/ngrammake_test.sh
+ mode change 100755 => 100644 src/test/ngrammarginalize_test.sh
+ mode change 100755 => 100644 src/test/ngrammerge_test.sh
+ mode change 100755 => 100644 src/test/ngramperplexity_test.sh
+ mode change 100755 => 100644 src/test/ngramprint_test.sh
+ mode change 100755 => 100644 src/test/ngramrand_test.sh
+ mode change 100755 => 100644 src/test/ngramrandgen_test.sh
+ mode change 100755 => 100644 src/test/ngramshrink_test.sh
+ mode change 100755 => 100644 src/test/ngramsymbols_test.sh
```

Comment:
I suspect that all these mode changes are unintentional (or at least unnecessary)?

Reply:
Good catch! I went through and cleaned up the patch files for this and other GRM builds.